### ISSUE TITLE
Get UK Target data

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/open-data-pvnet#2]
We would suggest using the UK PVlive national solar generation as target data. This can be retrieved by API and is fairly easy to use. 

Context: At OCF we actually predict GSP level solar generation and use a small ML model to sum it up. 

I suggest for this project we start simple, with just the UK national. Also note that the capacity changes, so we need to collect that too, which is possible via API above.

OCF might have a script for collecting this already, and can share. 

We should think of a place where we can store the data publicly. Becasue its time series data, its shouldn't be too big

**Comment by jcamier** [comment]
Potential data source:
https://www.metoffice.gov.uk/services/data/met-office-data-for-reuse/discovery

**Comment by peterdudfield** [comment]
> Potential data source: https://www.metoffice.gov.uk/services/data/met-office-data-for-reuse/discovery

yea this would be good for the NWP gridded data #3. For the target data we should use PVLive - https://www.solar.sheffield.ac.uk/pvlive/, they have an api and python package e.t.c

**Comment by jcamier** [comment]
Have you guys ever tried this API for PV predictions? I just found it today as I was researching our project...

https://joint-research-centre.ec.europa.eu/photovoltaic-geographical-information-system-pvgis/getting-started-pvgis/api-non-interactive-service_en

**Comment by peterdudfield** [comment]
We have. script here https://github.com/openclimatefix/uk-pv-backtest/blob/main/scripts/get_pvlive_data.py, that should collect the PVLive data. This might need modifitying so it only collections National Solar generation, i.e gsp_id=0

**Comment by jcamier** [comment]
This is great. Do you want us to use this library then to get the target data and maybe extend it or use this as a baseline in for a specific utility function in open data PVNet?

**Comment by peterdudfield** [comment]
Sorry, I'm not sure what you mean?

**Comment by jcamier** [comment]
There are two approaches we can take with open-data-pvnet:

1) Extend the uk-pv-backtest repository:

- We can enhance the features of the get_pvlive_data.py module or add a new module dedicated to collecting National Solar generation data.
- The enhanced or new module would then be included as a dependency in open-data-pvnet.
Develop directly within open-data-pvnet:

2) Create a new module within open-data-pvnet itself to handle the collection of National Solar generation data.
- This keeps everything self-contained within open-data-pvnet.

**Comment by peterdudfield** [comment]
I would go for 2. the uk-pv-bactest repo is what we use internal to generate our backtests and can change quite a lot. 
I would pull this script into open-data-pvnet and develop / tidy it up there

**Comment by jcamier** [comment]
@alirashidAR I added a review comment on your PR.

**Comment by alirashidAR** [comment]
> Thanks @alirashidAR for doing the PR. Would you like to collect the data too. We can then put it on an public s3 bucket ready for everyone to use

yes I would like to work on that . Thank you !

**Comment by alirashidAR** [comment]
@peterdudfield  
We would collect this data in the buckets and then create samples, right?  
And what should the time frame of the collected data be?

**Comment by peterdudfield** [comment]
2020 onwards? to be safe

**Comment by alirashidAR** [comment]
The target data should be stored in which format ?

**Comment by peterdudfield** [comment]
We tried in the past to stored it as netcdf file. 

Just to note it would be good to get the power generation, and also the capacity. The capacity can be got from PVLive too. This changes in time, as more PV systems come online

**Comment by sadavaidya** [comment]
Hello there,

I’m really interested in contributing to this project and would love to help with this issue of getting UK target data. Could you please confirm if this issue is still open and if there’s any existing work or specific guidelines I should be aware of before getting started?

I understand that @alirashidAR has already worked on a script to download the data in the required format using the Python API. I’d love to know how I can further contribute or build upon that work.

Looking forward to your guidance!

Best regards,
Sadashiv

**Comment by MAYANK12SHARMA** [comment]
Hi @peterdudfield, is there anything left to do? I’d be happy to do it?

**Comment by jcamier** [comment]
@alirashidAR I believe you are almost done with this issue. Can you let @MAYANK12SHARMA @sadavaidya if there any assistance you may need?

**Comment by MAYANK12SHARMA** [comment]
Hii @jcamier can you please take a moment for review the latest PR of mine for the issue #65

**Comment by MAYANK12SHARMA** [comment]
> [@alirashidAR](https://github.com/alirashidAR) I believe you are almost done with this issue. Can you let [@MAYANK12SHARMA](https://github.com/MAYANK12SHARMA) [@sadavaidya](https://github.com/sadavaidya) if there any assistance you may need?
Yes, @alirashidAR, please let me know if you need any assistance from my side.

**Comment by alirashidAR** [comment]
Hey @MAYANK12SHARMA @sadavaidya, this issue is almost complete. Thank you, but there isn't much left to work on right now.

**Comment by sadavaidya** [comment]
Thank you @alirashidAR , for letting us know. I will see if I can contribute to any other issue.

**Comment by jcamier** [comment]
@peterdudfield is this the dataset for PVlive?
https://huggingface.co/datasets/openclimatefix/uk_pv

I am curious if we should create another one with @alirashidAR open data pvnet changes that will be more suited for working with ocf-data-sampler ?

**Comment by peterdudfield** [comment]
> [@peterdudfield](https://github.com/peterdudfield) is this the dataset for PVlive? https://huggingface.co/datasets/openclimatefix/uk_pv
> 
> I am curious if we should create another one with [@alirashidAR](https://github.com/alirashidAR) open data pvnet changes that will be more suited for working with ocf-data-sampler ?

no that is site PV systems only. 

ocf-data-sampler is ready to use PVLive data. PVLive data is shared in s3 bucket already

**Comment by jcamier** [comment]
@peterdudfield when I go here:
aws s3 ls --no-sign-request s3://ocf-open-data-pvnet/data/uk/pvlive/v0/
We only have target.nc data. My understanding is all the data is supposed to be in .zarr.zip format?

**Comment by peterdudfield** [comment]
ah, you might have to convert it to a zarr zip then. Xarray should be able to do it

**Comment by peterdudfield** [comment]
The data is now `s3://ocf-open-data-pvnet/data/uk/pvlive/v2/combined_2023_gsp.zarr/`- this is for UK generation data




` f
